pkg/utils/reconciliationerrorhandler/error_handling.go

```diff
@@ -0,0 +1,140 @@
+package reconciliationerrorhandler
+
+import (
+	"fmt"
+)
+
+// GetID returns the ID of the error if possible.
+// If err does not implement ErrorID or is nill an empty string will be returned.
+func GetID(err error) string {
+	type errorIDer interface {
```

Comment:
I've seen this used in [this errors package][1] and some other places. 
The idea is that you force extracting the `id` only from the `GetID(err error)` method.

In our case this might be a bit too much limitation as there isn't any special logic (as in the cycle in [the error package][1])

[1]: https://github.com/pkg/errors/blob/master/errors.go#L270